thamos/discover.py

```diff
@@ -17,34 +17,72 @@
 
 """Implementation discovery methods to detect the current environment and its configuration."""
 
+import click
+import distro
 import os
 import sys
 import logging
-import typing
 import sysconfig
 
-from typing import Dict, Union
+from typing import Any
+from typing import Dict
+from typing import Optional
+from typing import Union
 
+from thoth.analyzer import run_command
 from thoth.common import map_os_name
 from thoth.common import normalize_os_version
 
-import distro
-import click
-from thoth.analyzer import run_command
 
 _LOGGER = logging.getLogger(__name__)
 _PROC_CPU_INFO = "/proc/cpuinfo"
 
 
-def discover_cuda_version(interactive: bool = False) -> typing.Optional[str]:
+# Taken from https://github.com/rpm-software-management/yum/blob/master/rpmUtils/miscutils.py as rpmUtils was deprecated
+def _split_rpm_filename(filename):
+    """
+    Pass in a standard style rpm fullname.
+
+    Return a name, version, release, epoch, arch, e.g.::
+        foo-1.0-1.i386.rpm returns foo, 1.0, 1, i386
+        1:bar-9-123a.ia64.rpm returns bar, 9, 123a, 1, ia64
+    """
+    if filename[-4:] == ".rpm":
+        filename = filename[:-4]
+
+    arch_index = filename.rfind(".")
+    arch = filename[arch_index + 1 :]
+
+    rel_index = filename[:arch_index].rfind("-")
+    rel = filename[rel_index + 1 : arch_index]
+
+    ver_index = filename[:rel_index].rfind("-")
+    ver = filename[ver_index + 1 : rel_index]
+
+    epoch_index = filename.find(":")
+    if epoch_index == -1:
+        epoch = ""
+    else:
+        epoch = filename[:epoch_index]
+
+    name = filename[epoch_index + 1 : ver_index].strip()
+    return name, ver, rel, epoch, arch
+
+
+def discover_cuda_version(interactive: bool = False) -> Optional[str]:
     """Check for CUDA version, if no CUDA is installed, return None."""
-    if bool(int(os.getenv("THAMOS_DISABLE_CUDA", 0))):
```

Comment:
What is the reasoning behind changing the default here?

Reply:
It seemed to me like a mistake in the first place, I think `THAMOS_DISABLE_CUDA` should be set to 1 instead of 0 to disable CUDA.

Reply:
I think that is already the case. CUDA is enabled by default and setting `THAMOS_DISABLE_CUDA` to `1` would disable it.

Reply:
Yes, so we should ensure that `THAMOS_DISABLE_CUDA` is disabled (i.e. set to 1) given the log message below?

Reply:
Only a value of 1 will enter the conditional, the 0 is the default value. If the env variable isn't present, then it will default to 1 and skip the conditional branch.